ch_tools/chadmin/internal/migration.py

```diff
@@ -78,6 +82,28 @@ def migrate_as_non_first_replica(ctx, database_name, temp_db):
     original_engine = metadata_non_repl_db.database_engine
     tables_info = _get_tables_info_and_detach(ctx, database_name)
 
+    # Unfortunately, it is not atomic.
+    # The schema could be changed before detach databases.
+    # However, it is sufficient for now, and we will stabilize
+    # this logic after we stop creating temporary tables.
+    for row in tables_info:
```

Comment:
We could not check the schema after detaching the database, because if the schema is different, we would not be able to reattach the temporary database (for example, on the second replica).